src/plugins/action/include/plugins/action/action.h

```diff
@@ -38,6 +38,8 @@ class Action : public PluginBase {
      */
     explicit Action(System& system);
 
+    explicit Action(std::shared_ptr<System> system);
```

Comment:
I'm not completely sure about the ownership story. So `Mavsdk` creates a list of `shared_ptr<System>`s that can outlive it. To each of those systems, `Mavsdk` gives a reference to a `unique_ptr<MavsdkImpl>`, meaning that a `System` must not outlive `Mavsdk`.

By giving a `System&` to the plugins, we were essentially saying: "Systems are owned by `Mavsdk`, so the plugins cannot outlive `Mavsdk` otherwise their reference is getting invalid". But by giving a `shared_ptr<System>`, we say "you own your destiny, do whatever you please", right? And that's not true, because a plugin cannot outlive `Mavsdk`, because each `System` still has a reference to `MavsdkImpl` :sweat_smile:.

Do I make any sense at all, or am I confused?